src/components/loading-indicator/LoadingIndicator.tsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import classNames from 'classnames';
+
+import { LoadingIndicatorPosition, LoadingIndicatorSize, LoadingIndicatorVisibility } from './types';
+
+import './LoadingIndicator.scss';
+
+export interface LoadingIndicatorProps {
+    /** Content to render in the component */
+    children?: React.ReactNode;
+    /** Custom CSS class for the component */
+    className?: string;
+    /** If `true`, the pulse animation is enabled */
+    isLoading?: boolean;
+    /** Vertical position of the component */
+    position?: LoadingIndicatorPosition;
+    /** Size of the pulse animation */
+    size?: LoadingIndicatorSize;
+    /** Visibility of the wrapped content */
+    visibility?: LoadingIndicatorVisibility;
+}
+
+const LoadingIndicator = ({
+    children,
+    className,
+    isLoading,
+    position = LoadingIndicatorPosition.MIDDLE,
+    size = LoadingIndicatorSize.REGULAR,
+    visibility = LoadingIndicatorVisibility.BLURRED,
+}: LoadingIndicatorProps) => {
+    if (isLoading === false) {
+        return children;
+    }
+
+    const classes = classNames('hui-LoadingIndicator', `hui-LoadingIndicator--${size}`, className, {
+        [`hui-LoadingIndicator--${position}`]: children,
+        [`hui-is-${visibility}`]: children,
+    });
+
+    // `isLoading` must not have a default value
+    const isEnabled = children ? isLoading : true;
```

Comment:
If there is wrapped content, the animation should be disabled by default.

If there is no wrapped content, the animation should be enabled by default.